src/components/application_manager/src/commands/hmi/sdl_activate_app_request.cc

```diff
@@ -55,9 +55,15 @@ void SDLActivateAppRequest::Run() {
       application_manager_.application(application_id);
 
   if (!app) {
```

Comment:
@anosach-luxoft, SDL has several list for application. First SDL have to search in list that contains already registered application, if SDL can't found it there, so it'll look at list that contains application that wait time when will be registered. 
There below, located code for this case.